lmdeploy/pytorch/engine/model_agent.py

```diff
@@ -1036,11 +1036,16 @@ def _construct(item):
             weights = ForkingPickler.loads(base64.b64decode(serialized_data))
             if request.load_format == 'flattened_bucket':
                 metadata: List[FlattenedTensorMetadata] = weights['metadata']
-                flattened_tensor: torch.Tensor = _construct(weights['flattened_tensor'])
-                bucket = FlattenedTensorBucket(flattened_tensor=flattened_tensor, metadata=metadata)
-                weights = bucket.reconstruct_tensors()
+                if metadata:
+                    flattened_tensor: torch.Tensor = _construct(weights['flattened_tensor'])
+                    bucket = FlattenedTensorBucket(flattened_tensor=flattened_tensor, metadata=metadata)
+                    weights = bucket.reconstruct_tensors()
+                else:
+                    # empty data
+                    weights = []
             else:
                 weights = [(k, _construct(v)) for k, v in weights]
+
```

Comment:
I guess if `request.load_format` is None, we shoud use the following code ?
```
bucket = FlattenedTensorBucket(named_tensors=weights)
weights = bucket.reconstruct_tensors()
```

This operation seems redundant and unnecessary. So I think `named_tensors` is useless in `FlattenedTensorBucket`.

Reply:
used in xtuner to create flatened_tensor as 
https://github.com/InternLM/xtuner/blob/eb371b86632fdceb5fab6a405412ced897a1cdbd/xtuner/v1/rl/base/worker.py#L847